libbeat/publisher/pipeline/output.go

```diff
@@ -167,7 +167,7 @@ func (w *netClientWorker) run() {
 
 func (w *netClientWorker) publishBatch(batch publisher.Batch) error {
 	ctx := context.Background()
-	if w.tracer != nil {
+	if w.tracer != nil && w.tracer.Active() {
```

Comment:
```suggestion
	if w.tracer != nil && w.tracer.Recording() {
```

Sorry, my fault - I pointed you towards the wrong method. We should use Recording, which can be dynamically updated (Active cannot).